packages/gatsby/src/internal-plugins/webpack-theme-component-shadowing/gatsby-node.js

```diff
@@ -0,0 +1,26 @@
+const GatsbyThemeComponentShadowingResolverPlugin = require(`.`)
+
+exports.onCreateWebpackConfig = ({
+  store,
+  stage,
+  getConfig,
+  rules,
+  loaders,
+  actions,
+}) => {
+  const { config, program } = store.getState()
+  const themes = config.__experimentalThemes.map(({ resolve }) => resolve)
```

Comment:
`internal-plugins` aren't auto added - they need to be declared in https://github.com/gatsbyjs/gatsby/blob/fb7abb5f4065fc7734c43d6257134c81eec63bfe/packages/gatsby/src/bootstrap/load-plugins/load.js#L154-L161 so checking if `config.__experimentalThemes` is defined can be done there. Maybe even preparing `themes` and passing it as pluginOption might make sense

Reply:
After checking out the other plugins, none of them use pluginOptions, but rather they use `getState()` as well and accept no options. So I moved the logic around so as to not cause undefined errors instead of passing options.